components/dashboard/src/provider-utils.tsx

```diff
@@ -97,6 +97,39 @@ async function openAuthorizeWindow(params: OpenAuthorizeWindowParams) {
     attachMessageListener(successKey, params);
 }
 
+async function redirectToAuthorize(params: OpenAuthorizeWindowParams) {
+    const { login, host, scopes, overrideScopes } = params;
+    const successKey = getUniqueSuccessKey();
+    const searchParamsReturn = new URLSearchParams({ message: successKey });
+    for (const [key, value] of new URLSearchParams(window.location.search)) {
+        if (key === "message") {
+            continue;
+        }
+        searchParamsReturn.append(key, value);
+    }
+    const returnTo = gitpodHostUrl
+        .with({ pathname: "/quickstart", search: searchParamsReturn.toString(), hash: window.location.hash })
```

Comment:
Should keep current path

Reply:
[`f86eaf8` (#19567)](https://github.com/gitpod-io/gitpod/pull/19567/commits/f86eaf88aa8b62a92769ef757e51e13d8121f08e)